src/cdk-experimental/testing/testbed.ts

```diff
@@ -0,0 +1,199 @@
+
+/**
+ * Component harness factory for testbed.
+ * @param componentHarness: Type of user defined harness.
+ * @param fixture: Component Fixture of the component to be tested.
+ */
+export function load<T extends ComponentHarness>(
+    componentHarness: ComponentHarnessConstructor<T>,
+    fixture: ComponentFixture<{}>): T {
+  const stabilize = async () => {
+    fixture.detectChanges();
+    await fixture.whenStable();
+  };
+  return new componentHarness(new UnitTestLocator(fixture.nativeElement, stabilize));
+}
+
+/**
+ * Gets the corresponding Element for the root of a TestElement.
+ */
+export function getNativeElement(testElement: TestElement): Element {
+  if (testElement instanceof UnitTestElement) {
+    return testElement.element;
+  }
+
+  throw Error(`Expected an instance of UnitTestElement, got ${testElement}`);
+}
+
+/**
+ * Locator implementation for testbed.
+ * Note that, this locator is exposed for internal usage, please do not use it.
+ */
+export class UnitTestLocator implements HarnessLocator {
+  private readonly _rootElement: TestElement;
+
+  constructor(private _root: Element, private _stabilize: () => Promise<void>) {
+    this._rootElement = new UnitTestElement(_root, this._stabilize);
+  }
+
+  host(): TestElement {
+    return this._rootElement;
+  }
+
+  async querySelector(selector: string, options?: QueryOptions): Promise<TestElement|null> {
+    await this._stabilize();
+    const e = getElement(selector, this._root, options);
+    return e && new UnitTestElement(e, this._stabilize);
+  }
+
+  async querySelectorAll(selector: string): Promise<TestElement[]> {
+    await this._stabilize();
+    return Array.prototype.map.call(
+        this._root.querySelectorAll(selector),
+        (e: Element) => new UnitTestElement(e, this._stabilize));
+  }
+
+  async load<T extends ComponentHarness>(
+    componentHarness: ComponentHarnessConstructor<T>, selector: string,
+    options?: QueryOptions): Promise<T|null> {
+    await this._stabilize();
+    const root = getElement(selector, this._root, options);
+    return root && new componentHarness(new UnitTestLocator(root, this._stabilize));
+  }
+
+  async loadAll<T extends ComponentHarness>(
+      componentHarness: ComponentHarnessConstructor<T>,
+      rootSelector: string): Promise<T[]> {
+    await this._stabilize();
+    return Array.prototype.map.call(
+        this._root.querySelectorAll(rootSelector),
+        (e: Element) => new componentHarness(new UnitTestLocator(e, this._stabilize)));
+  }
+}
+
+class UnitTestElement implements TestElement {
+  constructor(readonly element: Element, private _stabilize: () => Promise<void>) {}
+
+  async blur(): Promise<void> {
+    await this._stabilize();
+    triggerBlur(this.element as HTMLElement);
+    await this._stabilize();
+  }
+
+  async clear(): Promise<void> {
+    await this._stabilize();
+    if (!(this.element instanceof HTMLInputElement ||
+        this.element instanceof HTMLTextAreaElement)) {
```

Comment:
nit: I don't know how much we care about allowing people to run tests in Node, but angular/angular does it since it's faster. I believe that this will throw in Node. The safer way to check it would be to go through the `element.nodeName`.